src/components/chatItems.tsx

```diff
@@ -78,18 +48,45 @@ export function DevChatItem(props: {
       .map((action) => {
         return `${convertToRenderable(
           action.args,
-          functionNameToDisplay(action.name),
+          functionNameToDisplay(action.name)
         )}`;
       })
       .join("")}`;
   }
 
+  if (props.chatItem.role === "function") {
+    graphedData = extractGraphData(props.chatItem.content);
+    try {
+      const functionJsonResponse = JSON.parse(props.chatItem.content) as Json;
```

Comment:
Seems odd to me that `extractGraphData()` takes a string as input and then first lines are converting to JSON with try-catch when it's right next to another JSON-conversion with try-catch